nixos/modules/services/networking/pihole-ftl.nix

```diff
@@ -0,0 +1,102 @@
+{ pkgs
+, lib
+, config
+, ...
+}:
+
+with lib;
+
+let
+
+  cfg = config.services.pihole-ftl;
+  stateDir = "/etc/pihole";
+  confFile = "${stateDir}/pihole-FTL.conf";
+  logFile = "/var/log/pihole-FTL.log";
+
+in
+{
+
+  options.services.pihole-ftl = {
+    enable = mkEnableOption "Pi-hole FTL";
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Open ports in the firewall for Pi-hole FTL.";
+    };
+
+    config = mkOption {
```

Comment:
This doesn't seem right. This should probably be `types.lines`, and then the contents should be `pkgs.writeText`'d. Then you'd use the `L+` tmpfiles rule to link it into place.

Reply:
Do you have any examples of this setup? I can't seem to find a good one in nixpkgs.

Reply:
Here's roughly what I'd do (untested):

```diff
diff --git a/nixos/modules/services/networking/pihole-ftl.nix b/nixos/modules/services/networking/pihole-ftl.nix
index 02a49d415c7..77bbc1a47e8 100644
--- a/nixos/modules/services/networking/pihole-ftl.nix
+++ b/nixos/modules/services/networking/pihole-ftl.nix
@@ -12,6 +12,7 @@ let
   stateDir = "/etc/pihole";
   confFile = "${stateDir}/pihole-FTL.conf";
   logFile = "/var/log/pihole-FTL.log";
+  confText = pkgs.writeText "pihole-FTL.conf" cfg.config;
 
 in
 {
@@ -50,7 +51,7 @@ in
 
     systemd.tmpfiles.rules = [
       "d '${stateDir}' 0700 pihole pihole - -"
-      "f '${confFile}' 0400 pihole pihole - '${cfg.config}'"
+      "L+ '${confFile}' 0400 pihole pihole - '${confText}'"
       "f '${logFile}' 0700 pihole pihole - -"
     ];
 
```